servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/BeforeFinallyHttpOperator.java

```diff
@@ -303,149 +283,61 @@ public void cancel() {
 
         @Override
         public void onNext(@Nullable final Object o) {
-            if (!discardEventsAfterCancel) {
+            if (!discardEventsAfterCancel || state == PROCESSING_PAYLOAD) {
                 subscriber.onNext(o);
-                return;
-            }
-
-            boolean reentry = false;
```

Comment:
This `reentry` stuff was added in https://github.com/apple/servicetalk/commit/5d22672b817065a3b5e76bb12e0fa8355a998319. Tbh, I'm not sure what it's for, but that commit also broke the contract that the callbacks will be called exactly once. If we can sort out what it was there and if that was both important and isn't compensated for please let me know.

Reply:
Looking closer, it may be to prevent `beforeFinally.cancel()` occuring before a call to `subscriber.onNext(..)`. This seems unnecessary, but to minimize the changes in this PR I reverted the removal. We can address it later if we want.